.github/workflows/build-and-upload-staging.yml

```diff
@@ -2,69 +2,26 @@
 # SPDX-License-Identifier: Apache-2.0
 
 name: Build And Upload Staging Artifact
-env:
-  # Use terraform assume role for uploading to ecr
-  AWS_ASSUME_ROLE: ${{ secrets.TERRAFORM_AWS_ASSUME_ROLE }}
-  ECR_OPERATOR_STAGING_IMAGE: ${{ secrets.ECR_OPERATOR_STAGING_IMAGE }}
-
 on:
+  push:
+    branches:
+      - main
   workflow_dispatch:
-  workflow_call:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
 
 jobs:
-  MakeBinary:
-    name: 'MakeContainerImage'
-    runs-on: ubuntu-latest
+  BuildAndUpload:
+    uses: ./.github/workflows/build-and-upload-release.yml
+    secrets: inherit
     permissions:
       id-token: write
       contents: read
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v4
-        with:
-          go-version: ~1.19.6
-          cache: false
-
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v2
-        with:
-          role-to-assume: ${{ env.AWS_ASSUME_ROLE }}
-          aws-region: us-west-2
-
-      - name: Login to ECR
-        if: steps.cached_binaries.outputs.cache-hit == false
-        id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
-
-      - name: Set up Docker Buildx
-        if: steps.cached_binaries.outputs.cache-hit == false
-        uses: docker/setup-buildx-action@v1
-
-      - name: Set up QEMU
-        if: steps.cached_binaries.outputs.cache-hit == false
-        uses: docker/setup-qemu-action@v1
-
-      - name: Build Cloudwatch Agent Operator Image and push to ECR
-        uses: docker/build-push-action@v4
-        if: steps.cached_binaries.outputs.cache-hit == false
-        with:
-          file: ./Dockerfile
-          context: .
-          push: true
-          tags: ${{ env.ECR_OPERATOR_STAGING_IMAGE }}
-          platforms: linux/amd64, linux/arm64
-
-  e2e-test:
```

Comment:
we still want to run e2e tests on push to main to ensure we don't see a regression prior to release

Reply:
This happens in .github/workflows/build-and-upload-release.yml
Step "e2e-test"